test/OpenTelemetry.AutoInstrumentation.Tests/Configurations/PluginManagerTests.cs

```diff
@@ -150,6 +154,22 @@ public void ConfigureLogsOptionsSuccess()
         }
     }
 
+    [Fact]
+    public void ConfigureResourceSuccess()
+    {
+        var pluginAssemblyQualifiedName = typeof(MockPlugin).AssemblyQualifiedName!;
+        var settings = GetSettings(pluginAssemblyQualifiedName);
+        var pluginManager = new PluginManager(settings);
+
+        var resource = ResourceBuilder.CreateEmpty().InvokePlugins(pluginManager).Build();
+
+        using (new AssertionScope())
+        {
+            resource.Attributes.First().Key.Should().Be("key");
```

Comment:
@pellared Need your expertise here. Want to check of we could combine this test with `InvokeProviderPluginSuccess`? Tried created a ResourceBuilder mock, as the `AddAttribute` method takes `List<KeyValuePair>` verifiable was not happy with it. Hence created a new test, want to ensure I'm not missing anything here.

Reply:
I prefer this test test a lot more than `InvokeProviderPluginSuccess`. It does not use mocks, tests against "real" objects, therefore gives more trustworthiness. Moreover, for me it is easier to read.

However, personally, I would add a test to `IntegrationTests.PluginsTests` that would check the new functionality. You can add use in functionality to `TestApplication.Plugins.Plugin` and use the `collector.ResourceExpector` for asserting.

Reply:
Adding this to issue #1742 for adding integration test for this scenario.